src/server/routes/api/document/indra.js

```diff
@@ -91,27 +93,49 @@ const getDocuments = ( templates, queryArticle ) => {
       return [];
     }
 
+    const handleNoQueryAbastract = doc => {
+      if ( NO_ABSTRACT_HANDLING == 'text' ) {
+        return doc.toText();
+      }
+      else if ( NO_ABSTRACT_HANDLING == 'date' ) {
+        // no need to return a valid abstract since sorting will be based on date
+        sortByDate = true;
+        return null;
+      }
+      else {
+        throw `${NO_ABSTRACT_HANDLING} is not a valid value for NO_ABSTRACT_HANDLING environment variable!`;
+      }
+    };
+
+    const getAbstract = article => {
+      let abstract = article.MedlineCitation.Article.Abstract;
+      if ( _.isString( abstract ) ) {
+        return abstract;
+      }
+      if ( _.isArray( abstract ) ) {
+        return abstract[ 0 ];
+      }
+      return null;
+    };
+
+    const getPmid = article => article.PubmedData.ArticleIdList.find( o => o.IdType == 'pubmed' ).id;
```

Comment:
`PubmedData` may not exist: https://dtd.nlm.nih.gov/ncbi/pubmed/doc/out/180101/el-PubmedArticle.html

Reply:
In our case, the fetchPubmed will have `PubmedData: null`. I use lodash _.get for these cases.

Reply:
Okay, I updated the method in a safer way.